oxenc/bt_producer.h

```diff
@@ -299,19 +334,38 @@ class bt_list_producer {
         return bs->init + next + 1;
     }
 
+    template <string_like T>
+    void append(const T& data) {
+        if (has_child)
+            throw std::logic_error{"Cannot append to list when a sublist is active"};
+        append_impl(data.data(), data.size());
+        append_intermediate_ends();
+    }
+
     /// Appends an element containing binary string data
-    template <string_view_compatible T>
+    template <char_span_convertible T>
     void append(const T& data) {
         if (has_child)
             throw std::logic_error{"Cannot append to list when a sublist is active"};
-        append_impl(data);
+        append_impl(data.data(), data.size());
         append_intermediate_ends();
     }
 
-    template <string_view_compatible T>
-    bt_list_producer& operator+=(const T& data) {
-        append(data);
-        return *this;
+    /// Appends an element containing a const Char string
+    void append(std::string_view s) {
+        if (has_child)
+            throw std::logic_error{"Cannot append to list when a sublist is active"};
+        append_impl(s);
+        append_intermediate_ends();
+    }
+
+    /// Appends an element containing const_span data
```

Comment:
Comment seems incorrect: this function would accept any of `span<T>`, `const span<T>`, `span<const T>`, `const span<const T>`, but only the latter two of those is a `const_span`.